govc/library/checkout.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright (c) 2019 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/vapi/library"
+	"github.com/vmware/govmomi/vapi/library/finder"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/vcenter"
+)
+
+type checkout struct {
+	*flags.ClusterFlag
+	*flags.FolderFlag
+	*flags.ResourcePoolFlag
+	*flags.HostSystemFlag
+}
+
+func init() {
+	cli.Register("library.checkout", &checkout{}, true)
+}
+
+func (cmd *checkout) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.ClusterFlag, ctx = flags.NewClusterFlag(ctx)
+	cmd.ClusterFlag.Register(ctx, f)
+
+	cmd.ResourcePoolFlag, ctx = flags.NewResourcePoolFlag(ctx)
+	cmd.ResourcePoolFlag.Register(ctx, f)
+
+	cmd.HostSystemFlag, ctx = flags.NewHostSystemFlag(ctx)
+	cmd.HostSystemFlag.Register(ctx, f)
+
+	cmd.FolderFlag, ctx = flags.NewFolderFlag(ctx)
+	cmd.FolderFlag.Register(ctx, f)
+}
+
+func (cmd *checkout) Process(ctx context.Context) error {
+	if err := cmd.ClusterFlag.Process(ctx); err != nil {
+		return err
+	}
+	if err := cmd.ResourcePoolFlag.Process(ctx); err != nil {
+		return err
+	}
+	if err := cmd.HostSystemFlag.Process(ctx); err != nil {
+		return err
+	}
+	return cmd.FolderFlag.Process(ctx)
+}
+
+func (cmd *checkout) Usage() string {
+	return "PATH NAME"
+}
+
+func (cmd *checkout) Description() string {
+	return `Check out Content Library item PATH to vm NAME.
+
+Examples:
+  govc library.checkout -cluster my-cluster my-content/template-vm-item my-vm`
```

Comment:
Feels to me like the template parameter should maybe have a flag - multiple unflagged trailing parameters feel somewhat hard to reason about. But I'd prefer consistency with other options. Up to you.

Reply:
There are a few library commands, and a few others in govc that take 2 args when they are both required.  This one and the others are similar to the `cp` command: `cp SOURCE DEST`